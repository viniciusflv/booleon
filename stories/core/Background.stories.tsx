import { Background } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Background',
  component: Background,
};

export const Default: FC = () => (
  <Background
    gy_grey
    bg_black
    bg_img="https://www.w3schools.com/cssref/img_tree.gif"
    style={{ width: '500px', height: '1000px' }}>
    <span style={{ color: 'white' }}>texto</span>
  </Background>
);
