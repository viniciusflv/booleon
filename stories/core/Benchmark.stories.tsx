// import React, { useState, useEffect } from 'react';

// import { withKnobs } from '@storybook/addon-knobs';

// import { View } from '../../packages/base/src';
// import benchmarkProps from '../benchmarkProps';

// export default {
//   title: 'Core / Benchmark',
//   component: View,
//   decorators: [withKnobs],
// };

// const Div = (props) => <div {...props} />;

// const DivStyled = View.styled(Div);

// const Styled = ({ bool }) => (
//   <>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     {/* <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled> */}
//   </>
// );

// const Main = ({ bool }) => (
//   <>
//     <View.div content={true} {...benchmarkProps(bool)}>
//       div
//     </View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div>
//     {/* <View.div {...benchmarkProps(bool)}>div</View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div> */}
//   </>
// );

// const Multiple = ({ bool }) => (
//   <View.multiple {...benchmarkProps(bool)}>
//     <div>multiple</div>
//     <div>multiple</div>
//     <div>multiple</div>
//     {/* <div>multiple</div>
//     <div>multiple</div> */}
//   </View.multiple>
// );

// export const Default = () => {
//   const [bool, setBool] = useState(true);
//   useEffect(() => {
//     const timeout = setTimeout(() => setBool(!bool), 1000);
//     return () => clearTimeout(timeout);
//   }, [bool]);

//   return (
//     <>
//       <Multiple bool={bool} />
//       <Main bool={bool} />
//       <Styled bool={bool} />
//     </>
//   );
// };
