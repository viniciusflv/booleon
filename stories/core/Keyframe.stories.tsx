// import React, { FC } from 'react';

// import { withKnobs } from '@storybook/addon-knobs';

// import { View } from '../../packages/base/src';
// import { Keyframe } from '../../packages/base/src/Keyframe';

// export default {
//   title: 'Core / Keyframe',
//   component: View,
//   decorators: [withKnobs],
// };

// const Keyframes: FC<any> = ({ children }) => (
//   <>
//     <Keyframe.provider scale>
//       <Keyframe.state sl_50 />
//       <Keyframe.state sl_100 />
//     </Keyframe.provider>
//     <Keyframe.provider spin>
//       <Keyframe.state rt_360 />
//     </Keyframe.provider>
//     <Keyframe.provider turn>
//       <Keyframe.state rt_180 />
//     </Keyframe.provider>
//     <Keyframe.provider drop>
//       <Keyframe.state top />
//       <Keyframe.state top_full />
//     </Keyframe.provider>
//     <Keyframe.provider left>
//       <Keyframe.state left_neg />
//       <Keyframe.state left />
//     </Keyframe.provider>
//     <Keyframe.provider right>
//       <Keyframe.state right_neg />
//       <Keyframe.state right />
//     </Keyframe.provider>
//     {children}
//   </>
// );
// export const Default = () => {
//   return (
//     <Keyframes>
//       <View.div relative>
//         <View.multiple
//           h_50
//           w_50
//           ani_duration_3s
//           ani_iteration_infinite
//           ani_forwards
//           ani_ease_in_out>
//           <View.div bg_fd0f00 kf_scale />
//           <View.div bg_5f0f55 kf_spin />
//           <View.div bg_505f50 kf_turn />
//           <View.div bg_0d0f50 kf_left relative />
//           <View.div bg_0d5ff0 kf_right relative />
//           <View.div bg_0ffd5f kf_drop absolute />
//         </View.multiple>
//       </View.div>
//     </Keyframes>
//   );
// };
