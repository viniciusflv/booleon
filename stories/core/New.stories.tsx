import React from 'react';

import { styledBooleon, useTheme } from '../../packages/react/src';

import {
  spacing,
  sizing,
  background,
  border,
  shadow,
} from '../../packages/modules/src';

export default {
  title: 'Core / New',
};

const Div = styledBooleon.div(background, spacing, sizing, border, shadow);

export const Default = () => {
  const { toggleTheme } = useTheme();
  return (
    <Div
      sd_8
      h_400
      w_400
      b_edge
      btr_rounded
      bbl_rounded
      bg_d5d5d5
      dark__bg_2f2f2f
      onClick={toggleTheme}
    />
  );
};
