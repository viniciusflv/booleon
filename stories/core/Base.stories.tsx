import { Base } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Base',
  component: Base,
};

export const NoProps: FC<typeof Base> = () => <Base />;
