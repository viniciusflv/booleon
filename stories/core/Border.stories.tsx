import { Border } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Border',
  component: Border,
};

export const Default: FC = () => <Border />;
