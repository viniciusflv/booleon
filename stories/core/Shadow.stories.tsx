import { Container } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Container',
  component: Container,
};

export const Default: FC = () => (
  <div style={{ display: 'flex' }}>
    <div style={{ height: '1rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '2rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '4rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '8rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '16rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '32rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '64rem', margin: '10px', background: 'red', width: '10px' }} />
  </div>
);
