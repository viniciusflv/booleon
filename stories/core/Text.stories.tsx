import { Text } from '../../packages/core/src';
import { text, withKnobs } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import React, { FC } from 'react';

export default {
  title: 'Core | Text',
  component: Text,
  decorators: [withA11y, withKnobs],
};

export const Default: FC = () => {
  const props = text('Props', '')
    .split(/\s/)
    .reduce((acc, prop) => ({ ...acc, [prop]: true }), {});
  return <Text {...props}>{text('Text', 'Hello World')}</Text>;
};
