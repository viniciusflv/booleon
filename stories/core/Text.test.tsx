import 'jest-styled-components';
import { Text } from '../../packages/core/src/core';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Text', () => {
  describe('Style', () => {
    test('stroke', () => {
      const stroke = '-1px 0 #ff0, 0 1px #ff0, 1px 0 #ff0, 0 -1px #ff0';
      const component = renderer.create(<Text fs_ff0 />).toJSON();
      expect(component).toHaveStyleRule('text-shadow', stroke);
    });
  
    test('color', () => {
      const component = renderer.create(<Text fc_ff0 />).toJSON();
      expect(component).toHaveStyleRule('color', '#ff0');
    });
  
    test('italic', () => {
      const component = renderer.create(<Text f_italic />).toJSON();
      expect(component).toHaveStyleRule('font-style', 'italic');
    });
  });

  describe('Family', () => {
    test('ff_sans', () => {
      const ff_sans = 'Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif';
      const component = renderer.create(<Text f_sans />).toJSON();
      expect(component).toHaveStyleRule('font-family', ff_sans);
    });

    test('ff_serif', () => {
      const ff_serif = 'Georgia,Cambria,"Times New Roman",Times,serif';
      const component = renderer.create(<Text f_serif />).toJSON();
      expect(component).toHaveStyleRule('font-family', ff_serif);
    });

    test('ff_mono', () => {
      const ff_mono = 'Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace';
      const component = renderer.create(<Text f_mono />).toJSON();
      expect(component).toHaveStyleRule('font-family', ff_mono);
    });
  });

  describe('Size', () => {
    test('f_xs', () => {
      const component = renderer.create(<Text f_xs />).toJSON();
      expect(component).toHaveStyleRule('font-size', '1rem');
    });

    test('f_sm', () => {
      const component = renderer.create(<Text f_sm />).toJSON();
      expect(component).toHaveStyleRule('font-size', '2rem');
    });

    test('f_md', () => {
      const component = renderer.create(<Text f_md />).toJSON();
      expect(component).toHaveStyleRule('font-size', '3rem');
    });

    test('f_lg', () => {
      const component = renderer.create(<Text f_lg />).toJSON();
      expect(component).toHaveStyleRule('font-size', '4rem');
    });

    test('f_xl', () => {
      const component = renderer.create(<Text f_xl />).toJSON();
      expect(component).toHaveStyleRule('font-size', '5rem');
    });
  });
});
