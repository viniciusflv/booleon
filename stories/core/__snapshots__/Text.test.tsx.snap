// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text snapshot 1`] = `
.c0 {
  content: "*";
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  outline: none;
  -webkit-transition: easy .2s;
  transition: easy .2s;
  cursor: pointer;
  pointer-events: none;
  opacity: .1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  visibility: hidden;
  z-index: 666;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<div
  className="c0"
  content="\\"*\\""
  cursor="pointer"
  display="flex"
  opacity=".1"
  pointerEvents="none"
  visibility="hidden"
/>
`;
